charms/istio-pilot/src/charm.py

```diff
@@ -156,10 +158,13 @@ def handle_default_gateway(self, event):
         # Update the ingress resources as they rely on the default_gateway
         self.handle_ingress(event)
 
-        # check if gateway is created
-        self.handle_gateway_relation(event)
-
     def handle_gateway_relation(self, event):
+        if self.model.relations["gateway"]:
+            self.log.info(
+                "No gateway relation found, waiting for remote unit to join the relation"
```

Comment:
How come gateway is not found? Isn't the if statement checking for the relation's existence?